src/recipes/imgEdgeSearch.js

```diff
@@ -0,0 +1,117 @@
+                var scrollLeft = docEl.scrollLeft;
+                var clientTop = docEl.clientTop;
+                var clientLeft = docEl.clientLeft;
+                var width = box.width;
+                var height = box.height;
+                var top = box.top + scrollTop - clientTop;
+                var left = box.left + scrollLeft - clientLeft;
+                var bottom = top + height;
+                var right = left + width;
+                if(width == 0 || height == 0 || bottom <= 0 || right <= 0){
+                    return 0;
+                }
+            }          
+            //checks for visibility with computed style
+            var elStyle = getComputedStyle(element);
+            if(elStyle.getPropertyValue("display") === "none"){
+                return 0;
+            } 
+            else if(elStyle.getPropertyValue("opacity") < 0.1){
+                return 0;
+            } 
+            else if(elStyle.getPropertyValue("transform").includes(" 0,") || elStyle.getPropertyValue("transform").includes(" 0)")){
+                return 0;
+            } 
+            else if(elStyle.getPropertyValue("visibility") === "hidden"){
+                return 0;
+            }
+
+            // if text is within an iframe that does not appear
+            var elAbove = element;
+            do{
+                if(elAbove.nodeName === "IFRAME"){
+                    if(getComputedStyle(elAbove).getPropertyValue("src") === "" && getComputedStyle(elAbove).getPropertyValue("frameBorder") === 0){
+                        return 0;
+                    }
+                }
+                if(elAbove.parentElement !== null){
+                    elAbove = elAbove.parentElement;
+                }
+            } while(elAbove.parentElement !== null);
+            return 1;
+        }
+
+        //tests for images
+        if(element.nodeName == "IMG"){
+            var browsers = [{ str: (new RegExp("(internet(\\s|(\\-|\\_))?explorer|(^|\\W)ie($|\\W))", "gi")), name: "Internet Explorer" },
+            { str: (new RegExp("chrome[^b|$]", "gi")), name: "Chrome" },
+            { str: (new RegExp("firefox", "gi")), name: "Firefox" },
+            { str: (new RegExp("safari", "gi")), name: "Safari" },
+            { str: (new RegExp("(^|o|\\W)edge", "gi")), name: "Edge" },
+            { str: (new RegExp("(^|[o]|\\W)opera([^t]|\\W|$)", "gi")), name: "Opera" }];
+
+            for(var i = 0; i < browsers.length; i++){
+                if(element.getAttribute("alt") !== null){
+                    if(browsers[i].str.test(element.getAttribute("alt").toString())){
+                        var altMatch = element.getAttribute("alt").match(browsers[i].str);
+
+                        results[browsers[i].name] = results[browsers[i].name] || { count: 0, values: [] };
+                        results[browsers[i].name].count++;
+
+                        for(var j = 0; j < altMatch.length; j++){
```

Comment:
already have a loop on browsers above, so this would just double count "Chrome or Chrome Canary"